scripts/populate_tox/populate_tox.py

```diff
@@ -510,8 +585,40 @@ def main() -> None:
                     }
                 )
 
-    write_tox_file(packages)
+    if fail_on_changes:
+        old_file_hash = get_file_hash()
+
+    write_tox_file(
+        packages, update_timestamp=not fail_on_changes, last_updated=last_updated
+    )
+
+    if fail_on_changes:
+        new_file_hash = get_file_hash()
+        if old_file_hash != new_file_hash:
+            raise RuntimeError(
+                dedent(
+                    """
+                Detected an unexpected change in `tox.ini` that is not reflected
+                in `tox.jinja` and/or `populate_tox.py`.
```

Comment:
Can this error also occur if the jinja template and/or `populate_tox.py` script are changed, but we forget to regenerate the `tox.ini` file afterwards? If yes, please mention this possibility, as well